rpc/src/v1/impls/personal.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Account management (personal) rpc implementation
+use std::sync::{Arc, Weak};
+use jsonrpc_core::*;
+use v1::traits::Personal;
+use util::keys::store::*;
+use util::{Bytes, Address};
+use std::sync::RwLock;
+
+/// Account management (personal) rpc implementation.
+pub struct PersonalClient {
+	secret_store: Weak<SecretStore>,
+	unlocked_account: Arc<RwLock<Option<Address>>>,
+	unlocked_secret: Arc<RwLock<Option<Bytes>>>,
+}
+
+impl PersonalClient {
+	/// Creates new PersonalClient
+	pub fn new(store: &Arc<SecretStore>) -> Self {
+		PersonalClient {
+			secret_store: Arc::downgrade(store),
+			unlocked_account: Arc::new(RwLock::new(None)),
+			unlocked_secret: Arc::new(RwLock::new(None)),
+		}
+	}
+}
+
+impl Personal for PersonalClient {
+	fn accounts(&self, _: Params) -> Result<Value, Error> {
+		let store = take_weak!(self.secret_store);
+		match store.accounts() {
```

Comment:
minor issue. this function could be less lenient, and return error if there are any input params ;)